source/aspnetintegration.rst

```diff
@@ -263,3 +266,210 @@ A typical solution to this problem is to split up the class into multiple smalle
+
+The approach that will be laid out next is the preferred approach and it involves creating an adapter implementation that gets registered with the ASP.NET Core configuration system and forwards the calls to Simple Injector with the application-specific abstraction.
+
+The main idea with this approach is to have an application specific abstraction or abstractions which define the functionality to be performed in the IHostedService.  As an example, let's call this application-specific abstraction "IMyJob".  The implementation of IHostedService then becomes a simple adapter which can be used by ASP.NET Core to run application-specific dependencies in a hosted service similar to the way that the IControllerActivator abstraction implemented by SimpleInjectorControllerActivator provides an adapter for ASP.NET Core to be able to create Controllers with dependencies resolved by the Container.
+
+Given that, the IHostedService adapter might look like this:
+
+.. code-block:: c#
+
+    public class SimpleInjectorJobProcessorHostedService : IHostedService, IDisposable
+    {
+        private readonly Container _container;
+        private Timer _timer;
+
+        //Depending on how this adapter gets registered with the ASP.NET Core configuration system, this
+        //constructor may be called before all of the registrations have been added to the container.
+        public SimpleInjectorJobProcessorHostedService(Container container) => _container = container;
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            /*
+             * Now the adapter can resolve things from the container.
+             * For example, we could get a settings object from the container:
+             * ISettingsObj settings = _containter.GetInstance<ISettingsObj>();
+             * _timer = new Timer(this.DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(settings.TimerInterval));
+             */
+            _timer = new Timer(this.DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(5));
+            return Task.CompletedTask;
+        }
+
+        private void DoWork(object state)
+        {
+            // Run operation in a scope
+            using (AsyncScopedLifestyle.BeginScope(_container))
+            {
+                // Resolve the collection of IMyJob application-specific implementations and run them all
+                foreach (var service in _container.GetAllInstances<IMyJob>())
+                {
+                    service.DoWork();
+                }
+            }
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            _timer?.Change(Timeout.Infinite, 0);
+            return Task.CompletedTask;
+        }
+
+        public void Dispose() => _timer?.Dispose();
```

Comment:
Don't forget, all private fields are written without underscores.

Reply:
I went through and changed the rest of the examples to match the StyleCop guidelines.